package.json

```diff
@@ -5,18 +5,37 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
-    "lint": "next lint"
+    "lint": "next lint",
+    "typecheck": "tsc",
+    "format:check": "npx prettier -c \"**/*.{js,jsx,json,ts,tsx}\"   ",
+    "format": "npx prettier -w \"**/*.{js,jsx,json,ts,tsx}\"",
+    "prepare": "husky install",
+    "pre-commit": "lint-staged"
   },
   "dependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.16.0",
+    "@typescript-eslint/parser": "^5.16.0",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-react": "^7.29.4",
     "next": "12.0.10",
     "react": "17.0.2",
-    "react-dom": "17.0.2"
+    "react-dom": "17.0.2",
+    "typecheck": "tsc"
   },
   "devDependencies": {
     "@types/node": "17.0.14",
     "@types/react": "17.0.39",
     "eslint": "8.8.0",
     "eslint-config-next": "12.0.10",
-    "typescript": "4.5.5"
+    "lint-staged": ">=10",
+    "prettier": "2.6.1",
+    "typescript": "4.5.5",
+    "husky": "^7.0.0"
+  },
+  "lint-staged": {
+    "**/*.{js,jsx,json,ts,tsx}": [
+      "eslint",
+      "prettier --write"
+    ]
   }
 }
```

Comment:
Any reason we have got yarn.lock instead of package-lock.json?

Reply:
Because this project was built with Yarn that's why we have `yarn.lock`.